src/components/templates/TemplateWorkflowCard.vue

```diff
@@ -110,28 +111,80 @@ const getThumbnailUrl = (index = '') => {
       : api.apiURL(`/workflow_templates/${sourceModule}/${template.name}`)
 
   // For templates from custom nodes, multiple images is not yet supported
-  const indexSuffix = sourceModule === 'default' && index ? `-${index}` : ''
+  const indexSuffix = !!template.thumbnailVariant && index ? `-${index}` : ''
 
   return `${basePath}${indexSuffix}.${template.mediaSubtype}`
 }
 
+const thumbnailVariantRe = /.*\.(?<variant>[^\.]+)/
+const thumbnailVariantList = ['compareSlider', 'hoverDissolve', 'zoomHover']
+
+const getNameThumbnailVariant = (name: string) => {
+  const variant = name.match(thumbnailVariantRe)
+
+  if (variant === null) {
+    return null
+  }
+
+  if (
+    variant.groups.variant &&
+    thumbnailVariantList.indexOf(variant.groups.variant) > -1
+  ) {
+    console.log(`${variant.groups.variant} detected`)
+    return variant.groups.variant
+  }
+  return null
+}
+
+const tryCleanThumbnailVariant = (name: string) => {
+  const nameVariant = getNameThumbnailVariant(name)
+  if (nameVariant !== null) {
+    return name.replace(`.${nameVariant}`, '')
+  }
+  return name
+}
+
 const baseThumbnailSrc = computed(() =>
-  getThumbnailUrl(sourceModule === 'default' ? '1' : '')
+  getThumbnailUrl(template.thumbnailVariant ? '1' : '')
 )
 const overlayThumbnailSrc = computed(() =>
-  getThumbnailUrl(sourceModule === 'default' ? '2' : '')
+  getThumbnailUrl(template.thumbnailVariant ? '2' : '')
 )
 
+const thumbnailVariant = computed(() => {
+  if (template.thumbnailVariant !== undefined) {
+    return template.thumbnailVariant
+  }
+
+  const nameVariant = getNameThumbnailVariant(template.name)
+  if (nameVariant !== null) {
+    template.thumbnailVariant = nameVariant
```

Comment:
Here I update the template.thumbnailVariant. A bit dirty for an accessor, woudl you have a better suggestion ?